IPython/html/nbextensions.py

```diff
@@ -242,6 +242,62 @@ def install_nbextension(path, overwrite=False, symlink=False, user=False, prefix
             src = path
             _maybe_copy(src, full_dest, verbose)
 
+
+def make_cmdclass(setupfile, path, enable=None):
+    """Build nbextension cmdclass dict for the setuptools.setup method.
+
+    Parameters
+    ----------
+    setupfile: str
+        Path to the setup file.
+    path: str
+        Directory relative to the setup file that the nbextension code lives in.
+    enable: [str=None]
+        Extension to "enable".  Enabling an extension causes it to be loaded
+        automatically by the IPython notebook.
+
+    Usage
+    -----
+    setup(
+        name='flightwidgets',
+        ...
+        cmdclass=make_cmdclass(__file__, 'flightwidgets', 'flightwidgets/init'),
+    )
+    """
+
+    from setuptools.command.install import install
+    from setuptools.command.develop import develop
+    from os.path import dirname, abspath, join
+
+    from .services.config import ConfigManager
+
+    def run_nbextension_install(develop):
+        install_nbextension(join(dirname(abspath(setupfile)), path), symlink=develop)
+        if enable is not None:
+            print("Enabling the extension ...")
+            cm = ConfigManager()
+            cm.update('notebook', {"load_extensions": {enable: True}})
```

Comment:
I don't think that installation should = import, even in JavaScript

On Tue, Mar 17, 2015 at 5:50 PM, Min RK <notifications@github.com> wrote:

> In IPython/html/nbextensions.py
> <https://github.com/ipython/ipython/pull/7837#discussion_r26620108>:
>
> > +        cmdclass=make_cmdclass(__file__, 'flightwidgets', 'flightwidgets/init'),
> > +    )
> > +    """
> > +
> > +    from setuptools.command.install import install
> > +    from setuptools.command.develop import develop
> > +    from os.path import dirname, abspath, join
> > +
> > +    from .services.config import ConfigManager
> > +
> > +    def run_nbextension_install(develop):
> > +        install_nbextension(join(dirname(abspath(setupfile)), path), symlink=develop)
> > +        if enable is not None:
> > +            print("Enabling the extension ...")
> > +            cm = ConfigManager()
> > +            cm.update('notebook', {"load_extensions": {enable: True}})
>
> Fair enough. I guess it's debateable. More often than not, it's what the
> user wants, so maybe it's okay. It's not what I expect from installation in
> general, but perhaps extensions should be treated differently. For a pure
> Python extension, would you also expect the extension to be loaded by
> default for all IPython sessions? If not, what's the difference between the
> two?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ipython/ipython/pull/7837/files#r26620108>.
>



-- 
Brian E. Granger
Cal Poly State University, San Luis Obispo
@ellisonbg on Twitter and GitHub
bgranger@calpoly.edu and ellisonbg@gmail.com